fedmsg/crypto/x509.py

```diff
@@ -136,28 +139,38 @@ def fail(reason):
     #   validate_certificate will one day be a part of M2Crypto.SSL.Context
     #   https://bugzilla.osafoundation.org/show_bug.cgi?id=11690
 
-    default_ca_cert_loc = 'https://fedoraproject.org/fedmsg/ca.crt'
-    cafile = _load_remote_cert(
-        config.get('ca_cert_location', default_ca_cert_loc),
-        config.get('ca_cert_cache', '/etc/pki/fedmsg/ca.crt'),
-        config.get('ca_cert_cache_expiry', 0),
-        **config)
-
-    ctx = m2ext.SSL.Context()
-    ctx.load_verify_locations(cafile=cafile)
-    if not ctx.validate_certificate(cert):
-        return fail("X509 certificate is not valid.")
-
-    # Load and check against the CRL
-    crl = None
-    if 'crl_location' in config and 'crl_cache' in config:
-        crl = _load_remote_cert(
-            config.get('crl_location', 'https://fedoraproject.org/fedmsg/crl.pem'),
-            config.get('crl_cache', '/var/cache/fedmsg/crl.pem'),
-            config.get('crl_cache_expiry', 1800),
-            **config)
+    ca_location = config.get('ca_cert_location', 'https://fedoraproject.org/fedmsg/ca.crt')
+    crl_location = config.get('crl_location', 'https://fedoraproject.org/fedmsg/crl.pem')
+    fd, cafile = tempfile.mkstemp()
+    try:
+        ca_certificate, crl = utils.load_certificates(ca_location, crl_location)
+        os.write(fd, ca_certificate.encode('ascii'))
+        os.close(fd)
```

Comment:
You might want to move this to a finally block. If the ```write()``` raises an ```IOError``` nothing will close the file.

Reply:
Done